### PR TITLE
Refactor to generic Storage

### PR DESCRIPTION
**Pull request by onmyway133**
- This enables more generic usage for Codable, Image and other types with flexible transformers
- Ensure type safety
- No need for ImageWrapper https://github.com/hyperoslo/Cache/issues/174
- Use document directory by default https://github.com/hyperoslo/Cache/issues/179

### CODE REVIEW

**Comment by zenangst** [comment]
Using document directory won’t work on tvOS I think. You should test that.

**Comment by onmyway133** [comment]
@zenangst All tests have passed for both iOS and tvOS. Did you have a problem with tvOS ?

File: Source/Shared/Storage/DiskStorage.swift [hunk 1]
```diff
@@ -229,3 +236,16 @@ extension DiskStorage {
     }
   }
 }
+
+public extension DiskStorage {
+  func transform<U>(transformer: Transformer<U>) -> DiskStorage<U> {
+    // swiftlint:disable force_try
+    let storage = try! DiskStorage<U>(
```
Comment (vadymmarkov) [inline Source/Shared/Storage/DiskStorage.swift]: Is it safe to use force try here?

Comment (onmyway133) [reply Source/Shared/Storage/DiskStorage.swift]: I can improve this in the next PR

**Comment by vadymmarkov** [comment]
@onmyway133 I think using Document directory by default is not a good idea, I would leave cache. Read https://github.com/hyperoslo/Cache/issues/179#issuecomment-396909232. As for tvOS I think it won't work on the device.

**Comment by zenangst** [comment]
@onmyway133 the problem is on device not in the simulator. I ran into issues that the app would throw and exception and crash when trying to store files in the app directory. However, it worked just fine in the simulator and tests. Just to be sure, I’d install it on a device just to make sure that everything is dandy and fine. It worked with cache before as it used the apps cache directory instead of documents, which is allowed to modify.

**Comment by onmyway133** [comment]
@zenangst Yeah thanks for the advice. I've set back to using document directory. We have configuration with custom directory so I think that should be fine for user with more customised scenario.

**Pull request merged**